app/models/person.rb

```diff
@@ -28,11 +28,19 @@ class Person < ActiveRecord::Base
   validates :email, presence: true, unless: :phone
   validates :phone, presence: true, unless: :email
 
+  attr_accessor :address, :zip, :tags
+
   before_save :downcase_email
-  after_save :update_nation_builder
+  after_save :update_nation_builder, :save_location
 
   alias_method :location_association, :location
-  delegate :update_location, :zip_code, :city, :address_1, to: :location
+  delegate :update_location, :district, :state, to: :location
+
+  def self.create_or_update(person_params)
+    if email = person_params.delete(:email)
+      find_or_initialize_by(email: email).tap{|p| p.update(person_params)}
```

Comment:
Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.